utility/generateGetArtifactsNoTeamSelectedTrainingPhrases.js

```diff
@@ -0,0 +1,152 @@
+const fs = require('fs');
+const uuidv1 = require('uuid/v1');
+
+class TeamSuffix {
+  constructor() {
+    this.phrases = [' from team ', ' from the team '];
+  }
+
+
+  generateStrings() {
+    return this.phrases.map(x => (`
+    {
+        "text": "${x}",
+        "userDefined": false
+      },
+      {
+        "text": "Team",
+        "alias": "Team",
+        "meta": "@Team",
+        "userDefined": true
+      },`)).concat(['']);
+  }
+}
+class AuthorSuffix {
+  constructor() {
+    this.phrases = [' by ', ' uploaded by '];
+  }
+
+  generateStrings() {
+    return this.phrases.map(x => (`
+    {
+        "text": "${x}",
+        "userDefined": false
+      },
+      {
+        "text": "Arne",
+        "alias": "Author",
+        "meta": "@Author",
+        "userDefined": false
+      },`)).concat(['']);
+  }
+}
+class TagSuffix {
+  constructor() {
+    this.phrases = [' tagged with ', ' about ', ' tagged ', ' with tags ', ' with the tags '];
+  }
+
+  generateStrings() {
+    return this.phrases.map(x => (`
+    {
+        "text": "${x}",
+        "userDefined": false
+      },
+      {
+        "text": "tags",
+        "alias": "Tag",
+        "meta": "@sys.any",
+        "userDefined": true
+      },`)).concat(['']);
+  }
+}
+class DateSuffix {
+  constructor() {
+    this.phrases = [' from ', ' uploaded ', ' saved ', ' from the '];
+  }
+
+  generateStrings() {
+    return this.phrases.map(x => (`
+    {
+        "text": "${x}",
+        "userDefined": false
+      },
+      {
+        "text": "3 days ago",
+        "alias": "DatePeriod",
+        "meta": "@sys.date-period",
+        "userDefined": true
+      },`)).concat(['']);
+  }
+}
+
+class PrefixGenerator {
+  constructor() {
+    this.prefixes = ['Get me an image', 'I need an image', 'Show me an image'];
+  }
+
+  generateStrings() {
+    return this.prefixes.map(x => (`,
+    "data": [
+      {
+        "text": "${x}",
+        "userDefined": false
+      },`));
+  }
+}
+
+class SuffixGenerator {
+  constructor() {
+    this.suffixTypes = [new TeamSuffix(), new AuthorSuffix(), new DateSuffix(), new TagSuffix()];
+  }
+
+  generateAllPhrases() {
+    this.allSuffixes = [];
+    this.suffixTypes[0].generateStrings().forEach(
+      string1 => this.suffixTypes[1].generateStrings().forEach(
+        string2 => this.suffixTypes[2].generateStrings().forEach(
+          string3 => this.suffixTypes[3].generateStrings().forEach(
+            string4 => this.allSuffixes.push(string1 + string2 + string3 + string4),
+          ),
+        ),
+      ),
+    );
+    return this.allSuffixes;
+  }
+}
+
+class PhraseGenerator {
```

Comment:
This entire File is just code duplication. Please use some form of inheritance etc. to avoid this. You could just edit the 1st script to create 2 versions at once since it is unlikely someone will want to regenerate on without the other.

This is also a recipe for the intents getting out of synch.

Reply:
done